lib/template-factory.js

```diff
@@ -1,14 +1,21 @@
 'use strict';
 
-const componentTemplate = require ('../templates/component');
-const dumbComponentTemplate = require ('../templates/dumb-component');
-
-const getComponentTemplate = (componentName, isDumb, includedMethods) => {
-    if (isDumb) {
-        return dumbComponentTemplate(componentName);
-    } else {
-        return componentTemplate(componentName, includedMethods);
+const componentTemplate = require('../templates/component');
+const dumbComponentTemplate = require('../templates/dumb-component');
+const componentTestTemplate = require('../templates/component-test');
+
+const templateFactory = (template, ...args) => {
+    switch (template) {
+        case 'COMPONENT':
+            return componentTemplate(...args);
+        case 'DUMB_COMPONENT':
+            return dumbComponentTemplate(...args);
+        case 'COMPONENT_TEST':
+            return componentTestTemplate(...args);
+        default:
+            throw Error('A valid template string must be passed as the first arguement');
```

Comment:
In other parts of the app, rather than throw error's, I've been preferring to just console.error a message and then exit gracefully.  I'm not sold on that convention, necessarily, but curious what your thoughts are on that?

Reply:
I prefer throwing Errors for the simple fact that it facilitates deeper testing. As far as I'm aware there's no way to test if an Error is logged to the console other then by manually looking at the console. As long as the test assertions are working correctly, this Error should never be thrown in production so it's more of an Error to alert us that something went wrong. Let me know what you think?

Reply:
I certainly understand and don't disagree with your reasoning.  Good food for thought, so I'll certainly keep that in mind!  Thank you.